src/protocolsupport/protocol/serializer/ItemStackSerializer.java

```diff
@@ -102,7 +101,7 @@ public static void writeItemStack(ByteBuf to, ProtocolVersion version, String lo
 		}
 		NetworkItemStack witemstack = itemstack;
 		if (isToClient) {
-			witemstack = remapItemToClient(version, locale, witemstack.cloneItemStack());
```

Comment:
im thinking of changing my mind on this one.... was having lots of issues tracking down mutability of the itemstacks in various place... but i think this probably does make sense here, can at least guarantee that ItemStackSerializer wont mutate item stacks...